apps/mobile-wallet/src/features/buy/BuyModal.tsx

```diff
@@ -0,0 +1,138 @@
+import { Portal } from 'react-native-portalize'
+import { useSafeAreaInsets } from 'react-native-safe-area-context'
+import WebView, { WebViewNavigation } from 'react-native-webview'
+import styled, { useTheme } from 'styled-components/native'
+
+import AppText from '~/components/AppText'
+import Button from '~/components/buttons/Button'
+import BottomModal, { BottomModalProps } from '~/components/layout/BottomModal'
+import { ModalContent } from '~/components/layout/ModalContent'
+import ScreenTitle from '~/components/layout/ScreenTitle'
+import { useAppSelector } from '~/hooks/redux'
+import { InWalletTabsParamList } from '~/navigation/InWalletNavigation'
+import { selectDefaultAddress } from '~/store/addressesSlice'
+import { DEFAULT_MARGIN } from '~/style/globalStyle'
+
+interface BuyModalProps extends Omit<BottomModalProps, 'Content'> {}
+
+const BuyModal = (props: BuyModalProps) => {
+  const { t } = useTranslation()
+  const navigation = useNavigation<NavigationProp<InWalletTabsParamList>>()
+  const theme = useTheme()
+  const webViewRef = useRef<WebView>(null)
+  const insets = useSafeAreaInsets()
+  const defaultAddress = useAppSelector(selectDefaultAddress)
+  const [isDisclaimerAccepted, setIsDisclaimerAccepted] = useState(false)
+
+  const onAndroidBackPress = () => {
+    if (webViewRef.current) {
+      webViewRef.current.goBack()
+      return true // prevent default behavior (exit app)
+    }
+    return false
+  }
+
+  useEffect(() => {
+    if (Platform.OS === 'android') {
+      BackHandler.addEventListener('hardwareBackPress', onAndroidBackPress)
+      return () => {
+        BackHandler.removeEventListener('hardwareBackPress', onAndroidBackPress)
+      }
+    }
+  }, [])
+
+  const banxaURL =
+    'https://alephium.banxa-sandbox.com/' +
+    `?walletAddress=${defaultAddress?.hash}` +
+    `&theme=${theme.name}` +
+    `&backgroundColor=${theme.bg.primary.slice(1)}` +
+    `&textColor=${theme.font.primary.slice(1)}` +
+    `&primaryColor=${theme.global.accent}` +
+    `&secondaryColor=${theme.global.complementary}`
+
+  const handleModalCloseWhenFinished = (e: WebViewNavigation) => {
+    // Close modal if url equals default return URL confirgured in Banxa dashboard
+    if (e.url.includes('alephium.org')) {
+      navigation.navigate('TransfersScreen')
+      props.onClose()
+    }
+  }
+
+  return (
+    <Portal>
+      <BottomModal
+        {...props}
+        Content={(props) => (
+          <ModalContent {...props} contentContainerStyle={{ flex: 1, paddingTop: 0 }}>
+            {!isDisclaimerAccepted && (
+              <DisclaimerContent>
+                <ScreenTitle title={t('Disclaimer')} />
+                <AppText style={{ flex: 1 }}>
+                  {t(
+                    'You are about to access 3rd party services provided by Banxa.com through an in-app browser. Alephium does not control Banxa’s services. Banxa’s terms and conditions will apply, so please read and understand them before proceeding.'
```

Comment:
Wouldn't it be fair for the user to link them to the terms and conditions of Banxa here?

<img width="300" alt="image" src="https://github.com/user-attachments/assets/b5dc8d2e-af11-4243-8f4f-2b93ce88a238">

Reply:
The link to their T&C can change, so I'd rather keep it simple if possible

Reply:
Then maybe at list link to Banxa.com?